postgraas_server/management_resources.py

```diff
@@ -86,14 +86,27 @@ def delete(self, id):
                     host=current_app.postgraas_backend.master_hostname,
                     port=entity.port,
                     dbname=entity.db_name
-            ):
+            ) as conn:
```

Comment:
Unless I'm completely missing something, this statement is useless.

It **does** make sense to use a context manager, but you want to have your `try - except` **in there**.
This then should make your
```python
        finally:
            conn.close()
```
obsolete.

Reply:
Yes, the context manager should handle the closing.

Reply:
I wanted to leverage the **conn** object to check for existing sessions in the database, but it didn't make sense to do that in another try-except block. Please have a look to my new solution on my latest commit.

Important notes:

- I used explicit `closing` because apparently context managers don't work very well with some DB connections. Found [this](https://gist.github.com/bhtucker/7c63881dbf520e426975b1e5f9b6832c) reference about it. I tested it and it works
- I did not include the `if other_sessions > 0:` inside the upper try-except block because the `abort(409)` raises itself an exception that kept getting caught by the except portion and it produced weird results. That's why I opted to do the if and abort outside the block